routers/api/v1/admin/user.go

```diff
@@ -457,3 +460,82 @@ func GetAllUsers(ctx *context.APIContext) {
 	ctx.SetTotalCountHeader(maxResults)
 	ctx.JSON(http.StatusOK, &results)
 }
+
+// RenameUser api for renaming a user
+func RenameUser(ctx *context.APIContext) {
+	// swagger:operation POST /admin/users/{username}/rename admin adminRenameUser
+	// ---
+	// summary: Rename a user
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: username
+	//   in: path
+	//   description: existing username of user
+	//   type: string
+	//   required: true
+	// - name: body
+	//   in: body
+	//   required: true
+	//   schema:
+	//     "$ref": "#/definitions/RenameUserOption"
+	// responses:
+	//   "204":
+	//     "$ref": "#/responses/empty"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "422":
+	//     "$ref": "#/responses/validationError"
+
+	if ctx.ContextUser.IsOrganization() {
+		ctx.Error(http.StatusUnprocessableEntity, "", fmt.Errorf("%s is an organization not a user", ctx.ContextUser.Name))
+		return
+	}
+
+	newName := web.GetForm(ctx).(*api.RenameUserOption).NewName
+	// Check if user name has been changed
+	if ctx.ContextUser.LowerName != strings.ToLower(newName) {
+		if err := user_model.ChangeUserName(ctx.ContextUser, newName); err != nil {
+			switch {
+			case user_model.IsErrUserAlreadyExist(err):
+				ctx.Error(http.StatusUnprocessableEntity, "", ctx.Tr("form.username_been_taken"))
+			case user_model.IsErrEmailAlreadyUsed(err):
```

Comment:
How can a username change cause an `ErrEmailAlreadyUsed`? Wouldn't this be an internal server error instead?

Reply:
you're right. this line will never be reached. I've removed it :)